cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,255 @@
+      "  warnings.warn(errors.NumbaDeprecationWarning(msg))\n",
+      "/home/nfs/saljain/miniconda3/envs/fil-test/lib/python3.7/site-packages/numba/cuda/envvars.py:16: NumbaDeprecationWarning: \n",
+      "Environment variables with the 'NUMBAPRO' prefix are deprecated, found use of NUMBAPRO_LIBDEVICE=/usr/local/cuda/nvvm/libdevice.\n",
+      "\n",
+      "For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-numbapro-environment-variables\n",
+      "  warnings.warn(errors.NumbaDeprecationWarning(msg))\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "import pytest\n",
+    "import os\n",
+    "\n",
+    "from cuml import ForestInference\n",
+    "from cuml.test.utils import array_equal\n",
+    "from cuml.utils.import_utils import has_xgboost\n",
+    "\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "if has_xgboost():\n",
+    "    import xgboost as xgb\n",
+    "else:\n",
+    "    raise(\"Please install xgboost using the conda package,\"\n",
+    "          \" Use conda install -c conda-forge xgboost \"\n",
+    "          \"command to install xgboost\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### function that trains the xgboost model and performs prediction on it as well\n",
+    "\n",
+    "#### for additional information on the xgboost library please refer to the documentation on : \n",
+    "#### https://xgboost.readthedocs.io/en/latest/parameter.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_and_predict_xgboost_model(X_train, y_train,\n",
+    "                                    X_validation,\n",
+    "                                    y_validation,\n",
+    "                                    num_rounds, model_path):\n",
+    "    # set the xgboost model parameters\n",
+    "    params = {'silent': 1, 'eval_metric':'error',\n",
+    "              'objective':'binary:logistic',\n",
+    "              'max_depth': 25}\n",
+    "    dtrain = xgb.DMatrix(X_train, label=y_train)\n",
+    "    # train the xgboost model\n",
+    "    bst = xgb.train(params, dtrain, num_rounds)\n",
+    "\n",
+    "    # save the trained xgboost model\n",
+    "    bst.save_model(model_path)\n",
+    "\n",
+    "    # predict the xgboost model\n",
+    "    dvalidation = xgb.DMatrix(X_validation, label=y_validation)\n",
+    "    xgb_preds = bst.predict(dvalidation)\n",
+    "\n",
+    "    # convert the predicted values from xgboost into class labels\n",
+    "    xgb_preds = np.around(xgb_preds)\n",
+    "\n",
+    "    return xgb_preds, model_path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### setting the required parameters"
```

Comment:
I would use a 2nd level header `##` here. In all the other notebooks, I've just marked the header as `## Define Parameters`